src/gpu/intel/jit/ngen/ngen_config.hpp

```diff
@@ -30,5 +30,7 @@ using half = dnnl::impl::float16_t;
 
 #define NGEN_BFLOAT16_TYPE
 #define NGEN_HALF_TYPE
+#define NGEN_ENABLE_SOURCE_LOCATION true
```

Comment:
I think we should only enable source_location on debug builds; similarly for use of `__FILE__` below. oneDNN users may not want to leak internal directory paths into released applications/libraries (especially closed-source products).
```suggestion
#define NGEN_ENABLE_SOURCE_LOCATION (!defined(NDEBUG))
```

Reply:
The `__FILE__` macro is already used in many locations throughout oneDNN, for example, it is used in [common/verbose.hpp](https://github.com/oneapi-src/oneDNN/blob/main/src/common/verbose.hpp), [cpu/x64/jit_generator.hpp](https://github.com/oneapi-src/oneDNN/blob/main/src/cpu/x64/jit_generator.hpp). If this is a concern, it should be handled in a separate task addressing the library wide problem.

Reply:
@rjoursler -- fair enough, though I think that's the only place (well, that and the aarch64 equivalent) where directory paths would be visible in the binary outside of debug mode. Though LTO could remove them...

@vpirogov What would you like to do here?